cmd/snap/cmd_snap_op.go

```diff
@@ -113,8 +114,15 @@ type cmdInstall struct {
 }
 
 func (x *cmdInstall) Execute([]string) error {
+	var uuid string
+	var err error
+
 	cli := Client()
-	uuid, err := cli.InstallSnap(x.Positional.Snap, x.Channel)
+	if osutil.FileExists(x.Positional.Snap) {
+		uuid, err = cli.SideloadSnap(x.Positional.Snap)
```

Comment:
The daemon gets the entire file content via a POST right now. I guess thats not ideal and we can change that so that it gets a reference to the file only. This will indeed require to make the path absolute.